dart/lib/src/sentry_exception_factory.dart

```diff
@@ -57,11 +58,13 @@ class SentryExceptionFactory {
       }
     }
 
+    final throwableString = throwable.toString();
+
     // if --obfuscate feature is enabled, 'type' won't be human readable.
     // https://flutter.dev/docs/deployment/obfuscate#caveat
     return SentryException(
       type: (throwable.runtimeType).toString(),
-      value: throwable.toString(),
+      value: throwableString.isStackTrace() ? null : throwableString,
```

Comment:
Instead of setting it to `null`, we check if there's anything useful in the String that isn't the stack trace, and set that as a value, for example:

https://github.com/getsentry/sentry-dart/issues/1421#issuecomment-1536153484

The value in this case should be `NetworkError(type: NetworkErrorType.unknown, error: Instance of 'iH')` but everything else is discarded.

Reply:
So we now remove the numbered lines from the StackTrace, but i don't think we can remove the other data, as it ca pretty much be anything, right? Or is there a spec so we can parse out more line?

```
pid: 9437, tid: 10069, name 1.ui
os: android arch: arm64 comp: yes sim: no
build_id: 'bca64abfdfcc84d231bb8f1ccdbfbd8d'
isolate_dso_base: 752602b000, vm_dso_base: 752602b000
isolate_instructions: 7526344980, vm_instructions: 752633f000
```

Reply:
This is the header of the stack trace, so it should be gone as well.

I have the feeling that it could be as simple as:

Pseudocode:

```
final stStr = stackTrace.toString()
final exStr = exception.toString()
return exStr.replace(stStr, '').trim()
```

Does that work?